org.mwc.cmap.NarrativeViewer/src/org/mwc/cmap/NarrativeViewer/FilterTextCellRenderer.java

```diff
@@ -0,0 +1,471 @@
+        textLayout.setFont(item.getFont(getColumn()));
+
+        for (int i = 0; i < styleRanges.length; i++)
+        {
+          StyleRange curr = prepareStyleRange(styleRanges[i], true);
+          textLayout.setStyle(curr, curr.start, curr.start + curr.length - 1);
+        }
+      }
+
+      textLayout.draw(gc, getBounds().x + x, getBounds().y + textTopMargin
+          + topMargin);
+      textLayout.dispose();
+      textLayout = null;
+    }
+    else
+    {
+      if (textLayout == null)
+      {
+        textLayout = new TextLayout(gc.getDevice());
+        item.getParent().addDisposeListener(new DisposeListener()
+        {
+          public void widgetDisposed(DisposeEvent e)
+          {
+            if(textLayout!=null)
+              textLayout.dispose();
+          }
+        });
+      }
+
+      textLayout.setFont(gc.getFont());
+      String text = item.getText(getColumn());
+      textLayout.setText(text);
+      textLayout.setAlignment(getAlignment());
+      textLayout.setWidth(width < 1 ? 1 : width);
+      StyledString styledString = getFilterStyledString(text);
+      if (styledString != null)
+      {
+        StyleRange[] styleRanges = styledString.getStyleRanges();
+
+        textLayout.setText(text);
+        textLayout.setFont(item.getFont(getColumn()));
+
+        for (int i = 0; i < styleRanges.length; i++)
+        {
+          StyleRange curr = prepareStyleRange(styleRanges[i], true);
+          textLayout.setStyle(curr, curr.start, curr.start + curr.length - 1);
+        }
+      }
+
+      if (item.getParent().isAutoHeight())
+      {
+        // Look through all columns (except this one) to get the max height needed for this item
+        int columnCount = item.getParent().getColumnCount();
+        int maxHeight =
+            textLayout.getBounds().height + textTopMargin + textBottomMargin;
+        for (int i = 0; i < columnCount; i++)
+        {
+          GridColumn column = item.getParent().getColumn(i);
+          if (i != getColumn() && column.getWordWrap())
+          {
+            int height =
+                column.getCellRenderer().computeSize(gc, column.getWidth(),
+                    SWT.DEFAULT, item).y;
+            maxHeight = Math.max(maxHeight, height);
+          }
+        }
+
+        // Also look at the row header if necessary
+        if (item.getParent().isWordWrapHeader())
+        {
+          int height =
+              item.getParent().getRowHeaderRenderer().computeSize(gc,
+                  SWT.DEFAULT, SWT.DEFAULT, item).y;
+          maxHeight = Math.max(maxHeight, height);
+        }
+
+        if (maxHeight != item.getHeight())
+        {
+          item.setHeight(maxHeight);
+        }
+      }
+      textLayout.draw(gc, getBounds().x + x, getBounds().y + textTopMargin
+          + topMargin);
+      textLayout.dispose();
+      textLayout = null;
+    }
+
+    if (item.getParent().getLinesVisible())
+    {
+      if (isCellSelected())
+      {
```

Comment:
no it is comment from eclipse source [DefaultCellRenderer] they maybe had a thought on " user definable" but never added .